### PR TITLE
VG-2280 Fix delete

### PR DESCRIPTION
**Pull request by estrauser-ledger**
## What is this about?

Delete would take forever on huge accounts, we've optimized queries but we need to define indexes on libcore for it to be faster.

## Cute picture of lama
image
_Llama pending deletion_

### CODE REVIEW

File: src/main/scala/co/ledger/cria/domain/adapters/persistence/wd/WDTransactionRecordRepository.scala [hunk 1]
```diff
@@ -38,19 +40,33 @@ final class WDTransactionRecordRepository(
...
+          .transact(walletDaemon)
+      }
+      deletedCount <- IOUtils.withTimer("Delete Operations") {
+        WDTransactionQueries.deleteOperations(accountUid, blockHeight).transact(walletDaemon)
+      }
+      _ <- IOUtils.withTimer("Delete Inputs") {
+        WDTransactionQueries.deleteMarkedInputs.transact(walletDaemon)
+      }
+      _ <- IOUtils.withTimer("Delete Transactions") {
+        WDTransactionQueries.deleteMarkedTransactions.transact(walletDaemon)
```
Comment (cdiaz-ledger) [inline src/main/scala/co/ledger/cria/domain/adapters/persistence/wd/WDTransactionRecordRepository.scala:62]: I still don't love that this isn't transactional but what can we do 🤷🏻‍♂️

**Review by cdiaz-ledger — APPROVED** [review]
LGTM

File: src/main/scala/co/ledger/cria/domain/adapters/persistence/wd/WDTransactionRecordRepository.scala (hunk 1 above)
Comment (estrauser-ledger) [reply src/main/scala/co/ledger/cria/domain/adapters/persistence/wd/WDTransactionRecordRepository.scala:62]: Yes.. unfortunately. We tried but PG would take forever more time than with 4 transactions. With Jérémy's help with indexes it should go down to a few seconds and so should not be  too much of a problem. I was thinking, maybe the best would be to have a `$DELETION_STRING` that would incorporate a private information of the account (accoutnUid for ex), that would at least stop strange things from happening in the case of 2 reseeds at the same time

Comment (cdiaz-ledger) [reply src/main/scala/co/ledger/cria/domain/adapters/persistence/wd/WDTransactionRecordRepository.scala:62]: that makes sense. in that case we wouldnt try to delete other transactions 👍

**Review by cdiaz-ledger — APPROVED** [review]
LGTM again

File: src/it/scala/co/ledger/cria/CreateKeychainDev.scala [hunk 2]
```diff
@@ -36,9 +36,9 @@ class CreateKeychainDev extends AnyFlatSpec with DefaultContextLogging {
       keychain <- makeKeychainId(
         RegisterRequest(
           Xpub(
-            "xpub6DEHKg8fgKcb9at2u9Xhjtx4tXGyWqUPQAx2zNCzr41gQRyCqpCn7onSoJU4VS96GXyCtAhhFxErnG2pGVvVexaqF7DEfqGGnGk7Havn7C2"
+            "xpub6C3hkmDKqMgVR5JXnmj7g3Zm5fDdyfAqciXaTK6nubYHuwPZVyfq6mVPqRfe4kisVqZibNWgz7euv5EucEgHJCt7DjnoeVUrKoaqdoNQBsc"
```
[the remark this answers is not in the document]
Comment (estrauser-ledger) [reply src/it/scala/co/ledger/cria/CreateKeychainDev.scala:39]: right.......

**Pull request merged**